### PR TITLE
feat(DIZIWATCH): add presence 

### PR DESCRIPTION
**Pull request by emrovsky**
## Description 
A plugin for the movie/anime platform diziwatch.net

## Acknowledgements
- [✔️ ] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [✔️ ] I linted the code by running `yarn format`
- [✔️ ] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
image

### CODE REVIEW

File: websites/D/diziwatch/metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		"id": "510174958270545920",
+		"name": "emrovsky"
+	},
+	"service": "diziwatch",
+	"description": {
+		"en": "diziwatch is a platform you can watch animes,movies"
+	},
+	"url": "diziwatch.net",
+	"version": "1.0.0",
+	"logo": "https://files.sikayetvar.com/lg/cmp/27/274975.png",
```
Comment (theusaf) [inline websites/D/diziwatch/metadata.json:13]: Logos should be a 512x512px square

File: websites/D/diziwatch/presence.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const enum Assets {
+	Logo = "https://i.imgur.com/45R11f9.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+	};
+	switch (document.URL) {
```
Comment (theusaf) [inline websites/D/diziwatch/presence.ts:14]: Prefer `document.location.href` over `document.URL`. 
However, for the use case here, `document.location.pathname` may be a better choice for maintainability.

File: websites/D/diziwatch/presence.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+					).textContent,
+					animeImg = document
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-preview.jw-reset"
+						)
+						.getAttribute("style")
+						.split('"')[1],
+					playStatus = document
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-display.jw-reset > div > div > div.jw-display-icon-container.jw-display-icon-display.jw-reset > div"
```
Comment (theusaf) [inline websites/D/diziwatch/presence.ts:58]: If possible, try to use less specific selectors. (Check for other properties or elements to access these elements)

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/D/diziwatch/metadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.10",
+	"author": {
+		"id": "510174958270545920",
+		"name": "emrovsky"
+	},
+	"service": "diziwatch",
+	"description": {
+		"en": "diziwatch is a platform you can watch animes,movies"
+	},
+	"url": "diziwatch.net",
```
Comment (theusaf) [inline websites/D/diziwatch/metadata.json:11]: There are new fields called `matches` and `iFrameMatches`, which are required for new Presences.
These are an array of strings which need to follow https://developer.chrome.com/docs/extensions/develop/concepts/match-patterns. 

```suggestion
	"url": "diziwatch.net",
	"matches": ["*://diziwatch.net/*"],
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/D/diziwatch/presence.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+	};
+	switch (document.URL) {
+		case "https://diziwatch.net/": {
+			presenceData.details = "Ana sayfa görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:17]:
```suggestion
```

File: websites/D/diziwatch/presence.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/45R11f9.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:12]:
```suggestion
		largeImageKey: Assets.Logo,
		startTimestamp: browsingTimestamp,
```

File: websites/D/diziwatch/presence.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	switch (document.URL) {
+		case "https://diziwatch.net/": {
+			presenceData.details = "Ana sayfa görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
+
+			break;
+		}
+		case "https://diziwatch.net/calendar/": {
+			presenceData.details = "Takvim sayfası görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:23]:
```suggestion
```

File: websites/D/diziwatch/presence.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+			break;
+		}
+		case "https://diziwatch.net/anime-arsivi/": {
+			presenceData.details = "Anime arşivi görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:29]:
```suggestion
```

File: websites/D/diziwatch/presence.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+						.textContent
+				} inceleniyor`;
+				presenceData.startTimestamp = browsingTimestamp;
+			} else {
+				const currentTime = document.querySelector(
+						"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-controlbar.jw-reset > div.jw-reset.jw-button-container > div.jw-icon.jw-icon-inline.jw-text.jw-reset.jw-text-elapsed"
+					).textContent,
+					duration = document.querySelector(
+						"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-controlbar.jw-reset > div.jw-reset.jw-button-container > div.jw-icon.jw-icon-inline.jw-text.jw-reset.jw-text-duration"
+					).textContent,
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:46]:
```suggestion
				const video = document.querySelector('video'),
```

File: websites/D/diziwatch/presence.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-preview.jw-reset"
+						)
+						.getAttribute("style")
+						.split('"')[1],
+					playStatus = document
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-display.jw-reset > div > div > div.jw-display-icon-container.jw-display-icon-display.jw-reset > div"
+						)
+						.getAttribute("aria-label");
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:60]:
```suggestion
					playStatus = video.paused
```

File: websites/D/diziwatch/presence.ts [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+				presenceData.details = videoTitle;
+
+				if (playStatus === "Oynat")
+					presenceData.state = `Duraklatıldı, ${currentTime} / ${duration}`;
+
+				if (playStatus === "Duraklat")
+					presenceData.state = `İzleniyor, ${currentTime} / ${duration}`;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:68]:
```suggestion
				if (playStatus) [, presenceData.endTimestamp] = presence.getTimestampsfromMedia(video)
				presenceData.smallImageKey = playStatus ? Assets.Paused : Assets.Play
				
```

File: websites/D/diziwatch/presence.ts [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+				presenceData.largeImageKey = animeImg;
+				presenceData.largeImageText = document.querySelector(
+					"#benzerli > div > span"
+				).textContent;
+
+				presenceData.smallImageKey = Assets.Logo;
+				presenceData.smallImageText = "Diziwatch";
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:76]:
```suggestion
```

File: websites/D/diziwatch/presence.ts [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+				presenceData.startTimestamp = browsingTimestamp;
+			} else {
+				const currentTime = document.querySelector(
+						"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-controlbar.jw-reset > div.jw-reset.jw-button-container > div.jw-icon.jw-icon-inline.jw-text.jw-reset.jw-text-elapsed"
+					).textContent,
+					duration = document.querySelector(
+						"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-controlbar.jw-reset > div.jw-reset.jw-button-container > div.jw-icon.jw-icon-inline.jw-text.jw-reset.jw-text-duration"
+					).textContent,
+					videoTitle = document.querySelector(
+						"#player > div.jw-wrapper.jw-reset > div.jw-title.jw-reset-text > div.jw-title-primary.jw-reset-text"
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:48]:
```suggestion
						".title"
```

File: websites/D/diziwatch/presence.ts [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+					).textContent,
+					animeImg = document
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-preview.jw-reset"
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:52]:
```suggestion
							".jw-preview.jw-reset"
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/D/diziwatch/metadata.json (hunk 1 above)
Comment (DarkVillager) [inline websites/D/diziwatch/metadata.json:13]:
```suggestion
	"logo": "https://i.imgur.com/p9ktkQw.png",
```

File: websites/D/diziwatch/presence.ts [hunk 15]
```diff
@@ -0,0 +1,81 @@
+const presence = new Presence({
+		clientId: "1214974615202299994",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/45R11f9.png",
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:7]:
```suggestion
	Logo = "https://i.imgur.com/p9ktkQw.png",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
Logos must be 512x512 like this:

File: websites/D/diziwatch/presence.ts [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+					playStatus = document
+						.querySelector(
+							"#player > div.jw-wrapper.jw-reset > div.jw-controls.jw-reset > div.jw-display.jw-reset > div > div > div.jw-display-icon-container.jw-display-icon-display.jw-reset > div"
+						)
+						.getAttribute("aria-label");
+
+				presenceData.details = videoTitle;
+
+				if (playStatus) [, presenceData.endTimestamp] = presence.getTimestampsfromMedia(video)
+				presenceData.smallImageKey = playStatus ? Assets.Paused : Assets.Play
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:58]: OOps
```suggestion
				presenceData.smallImageKey = playStatus ? Assets.Pause : Assets.Play
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by emrovsky** [comment]
How about this? @DarkVillager

File: websites/D/diziwatch/presence.ts (hunk 8 above)
Comment (DarkVillager) [reply websites/D/diziwatch/presence.ts:29]: This has not been resolved.

File: websites/D/diziwatch/presence.ts (hunk 7 above)
Comment (DarkVillager) [reply websites/D/diziwatch/presence.ts:23]: This has not been resolved.

File: websites/D/diziwatch/presence.ts (hunk 5 above)
Comment (DarkVillager) [reply websites/D/diziwatch/presence.ts:17]: This has not been resolved.

File: websites/D/diziwatch/presence.ts (hunk 6 above)
Comment (DarkVillager) [reply websites/D/diziwatch/presence.ts:12]: This has not been resolved.

File: websites/D/diziwatch/presence.ts [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+		}
+		case "/dizi-arsivi/": {
+			presenceData.details = "Dizi arşivi görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
+
+			break;
+		}
+		case "/episodes/": {
+			presenceData.details = "Bölümler inceleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:41]:
```suggestion
```

File: websites/D/diziwatch/presence.ts [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+		}
+		case "/anime-arsivi/": {
+			presenceData.details = "Anime arşivi görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
+
+			break;
+		}
+		case "/dizi-arsivi/": {
+			presenceData.details = "Dizi arşivi görüntüleniyor";
+			presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:35]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/D/diziwatch/presence.ts (hunk 6 above)
Comment (theusaf) [reply websites/D/diziwatch/presence.ts:12]: Simplify the code/remove duplication by placing the `startTimestamp` field in the initial data.

File: websites/D/diziwatch/presence.ts [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+			break;
+		}
+		case "/contact-us/": {
+			presenceData.details = "İletişim bilgileri inceleniyor 💀";
+			presenceData.startTimestamp = browsingTimestamp;
+
+			break;
+		}
+		default:
+			if (document.location.href.startsWith("https://diziwatch.net/dizi/")) {
```
Comment (theusaf) [inline websites/D/diziwatch/presence.ts:52]:
```suggestion
			if (document.location.pathname.startsWith("/dizi/")) {
```

File: websites/D/diziwatch/presence.ts (hunk 11 above)
Comment (theusaf) [reply websites/D/diziwatch/presence.ts:82]: It is a good idea to use timestamps and small images to display the current playing state.

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

File: websites/D/diziwatch/presence.ts [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+			break;
+		}
+		default:
+			if (document.location.pathname.startsWith("/dizi/")) {
+				presenceData.details = `${
+					document.querySelector("#content > div.incontentx > div.title > h1")
+						.textContent
+				} inceleniyor`;
+			} else {
+				presenceData.startTimestamp = null;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:48]: What

File: websites/D/diziwatch/presence.ts [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+
+const enum Assets {
+	Logo = "https://i.imgur.com/RMrW6j3.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+	};
+	presenceData.startTimestamp = browsingTimestamp;
```
Comment (DarkVillager) [inline websites/D/diziwatch/presence.ts:14]: What

**Pull request merged**